diff --git a/configure b/configure
index eb6719b..f7982e2 100755
--- a/configure
+++ b/configure
@@ -520,7 +520,7 @@ case $PLATFORM in
 				CFLAGS="$CFLAGS -march=armv6k";
 				;;
 			"armv7"|"armv7l")
-				CFLAGS="$CFLAGS -march=armv7-a";
+				CFLAGS="$CFLAGS -march=arm";
 				;;
 		esac
 		RTM_ENABLE="CK_MD_RTM_DISABLE"
@@ -706,23 +706,14 @@ int main(void) {
 }
 EOF
 
-$CC -o .1 .1.c
-COMPILER=`./.1 2> /dev/null`
-r=$?
-rm -f .1.c .1
-
-if test "$r" -ne 0; then
-	assert "" "update compiler"
-else
-	echo "success [$CC]"
-fi
+echo "success [$CC]"
 
 if test "$COMPILER" = "suncc"; then
 	LD=/bin/ld
 	LDFLAGS="-G -z text -h libck.so.$VERSION_MAJOR $LDFLAGS"
 	CFLAGS="-xO5 $CFLAGS"
 	PTHREAD_CFLAGS="-mt -lpthread"
-elif test "$COMPILER" = "gcc" || test "$COMPILER" = "clang" || test "$COMPILER" = "mingw32" || test "$COMPILER" = "mingw64"; then
+else
 	LD=$CC
 	SONAME="$LDNAME_MAJOR"
 	if test "$SYSTEM" = "darwin"; then
@@ -777,8 +768,6 @@ elif test "$COMPILER" = "gcc" || test "$COMPILER" = "clang" || test "$COMPILER"
 	if test "$SYSTEM" != "dragonflybsd"; then
 		CFLAGS="$CFLAGS -D_XOPEN_SOURCE=600"
 	fi
-else
-	assert "" "unknown compiler"
 fi
 
 printf "Finding suitable archiver........"
